grunt/config/browserify.js

```diff
@@ -7,11 +7,30 @@ var grunt = require('grunt');
 var UglifyJS = require('uglify-js');
 var uglifyify = require('uglifyify');
 var derequire = require('derequire');
+var aliasify = require('aliasify');
+var globalShim = require('browserify-global-shim');
 var collapser = require('bundle-collapser/plugin');
 
 var envifyDev = envify({NODE_ENV: process.env.NODE_ENV || 'development'});
 var envifyProd = envify({NODE_ENV: process.env.NODE_ENV || 'production'});
 
+var SECRET_INTERNALS_NAME = 'React.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED';
+
+var shimSharedModules = globalShim.configure({
+  './ReactCurrentOwner': SECRET_INTERNALS_NAME + '.ReactCurrentOwner',
+  './ReactComponentTreeHook': SECRET_INTERNALS_NAME + '.ReactComponentTreeHook',
+  // All these methods are shared are exposed.
+  './ReactElement': 'React',
+  './ReactPropTypes': 'React.PropTypes',
+  './ReactChildren': 'React.Children',
+});
+
+var shimDOMModules = aliasify.configure({
+  'aliases': {
+    './ReactAddonsDOMDependencies': './build/modules/ReactAddonsDOMDependenciesUMDShim.js',
```

Comment:
I noticed the shim part, was asking about the RHS :)

I looked at docs though - let's make that `{relative: './ReactAddonsDOMDependenciesUMDShim'}` and it'll work and be a bit more future proof (tested locally, it works).